sources/engine/Stride.Input/Windows/InputSourceWindowsDirectInput.cs

```diff
@@ -54,12 +60,24 @@ public override void Update()
             devicesToRemove.Clear();
         }
 
+        /// <summary>
+        /// Select all device IDs that contain "IG_".  If so, it's an XInput device
+        /// This information can not be found from DirectInput 
+        /// </summary>
+        private IEnumerable<CimInstance> GetAllXInputDevices()
+        {
+            return mySession.QueryInstances(@"root\cimv2", "WQL", "SELECT DeviceID FROM Win32_PNPEntity WHERE DeviceID LIKE '%&IG_%'");
+        }
```

Comment:
Consider returning a list of strings here:
```csharp
var query = mySession.QueryInstances(...);
var deviceIdPrefixes = query.Select(device => regex.Match(device.CimInstanceProperties["DeviceID"].Value.ToString()))
    .Where(match => match.Success)
    .Select(match => (match.Groups[1].ToString() + match.Groups[2].ToString()).ToLower())
    .ToList();
```
Then in `IsXInputDevice(Guid productGuid)` all you need to do is loop over prefixes and check
```csharp
if (productGuid.ToString().StartsWith(guidPart))
{
     return true;
}
```

Reply:
done! Also I think ToList() is unnecessary.

Reply:
Without the ToList you will repeat regex matching on each enumeration (if the query returns a materialized enumerable).

Reply:
right! then I'll add it too